test/test.py

```diff
@@ -99,6 +102,61 @@ def test(dirname=None, config_path=None, force=False):
                 config_path,
                 '\n'.join(diff_lines)))
 
+def test_delay():
+    """
+    Test that r2e waits before fetching repeatedly from the same
+    server.
+    """
+    delay_cfg = """[DEFAULT]
+    to = example@example.com
+    same-server-fetch-interval = 1
+    """
+    tmpdir = tempfile.mkdtemp()
+    cfg_path = _os.path.join(tmpdir, "rss2email.cfg")
+    data_path = _os.path.join(tmpdir, "rss2email.json")
+
+    def r2e(*args):
+        subprocess.call(["r2e", "-c", cfg_path, "-d", data_path] + list(args))
+
+    with open(cfg_path, "w") as f:
+        f.write(delay_cfg)
+
+    num_requests = 3
+
+    def webserver(queue):
+        import http.server, time
+
+        class NoLogHandler(http.server.SimpleHTTPRequestHandler):
+            def log_message(self, format, *args):
+                return
+
+        httpd = http.server.HTTPServer(('', 8000), NoLogHandler)
```

Comment:
Can you instead put a random unused port here (passing `0` usually does the trick), so that the test doesn’t fail if a user has something listening on 8000.

Reply:
Good catch, I just addressed this.